### PR TITLE
[PT] Support custom modules in PTQ

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

- Wrap model with tracing of parameters for PTQ.
- `PTModelExtractionWithFusedBiasCommand` to `PTModelExtractionCommand`
- Removed model_analyzer.py
- Renamed PTDepthwiseConv3dSubtype to PTModuleDepthwiseConv3dSubtype metatype
- Added PTModuleDepthwiseConv3dSubtype
- Added is_subtype to OperatorMetatypeRegistry.register  

### Reason for changes

Support models with custom modules.

### Related tickets

129581

### CODE REVIEW

**Comment by AlexanderDokuchaev** [comment]
manual/job/post_training_quantization/345 - with replace_timm_custom_modules_with_torch_native
manual/job/post_training_quantization/344/artifact/results.html - w/o replace_timm_custom_modules_with_torch_native
nightly/job/TriggerBetta/656 - (1 model failed by not related issue)

File: nncf/torch/quantization/strip.py [hunk 1]
```diff
@@ -139,23 +139,24 @@ def remove_disabled_quantizers(model: NNCFNetwork) -> NNCFNetwork:
...
-                if isinstance(op, BaseQuantizer) and not op.is_enabled_quantization():
-                    nncf_module.remove_pre_forward_operation(key)
-
-        if hasattr(nncf_module, "post_ops"):
-            for key in list(nncf_module.post_ops.keys()):
-                op = nncf_module.post_ops(key)
-                if isinstance(op, BaseQuantizer) and not op.is_enabled_quantization():
-                    nncf_module.remove_post_forward_operation(key)
+    if model.nncf.replace_modules:
+        for node in model.nncf.get_original_graph().get_all_nodes():
```
Comment (daniil-lyakhov) [inline nncf/torch/quantization/strip.py:143]:
```suggestion
    if not  model.nncf.replace_modules:
         return model
    ...
```

**Comment by AlexanderDokuchaev** [comment]
manual/job/post_training_quantization/349/

File: nncf/quantization/algorithms/min_max/torch_backend.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ class PTMinMaxAlgoBackend(MinMaxAlgoBackend):
 
     @property
     def mat_mul_metatypes(self) -> List[OperatorMetatype]:
-        return [om.PTModuleLinearMetatype, om.PTLinearMetatype, om.PTMatMulMetatype]
+        return [om.PTLinearMetatype, om.PTLinearMetatype, om.PTMatMulMetatype]
```
Comment (daniil-lyakhov) [inline nncf/quantization/algorithms/min_max/torch_backend.py:63]: Please remove a duplication in the list

File: nncf/torch/model_transformer.py [hunk 3]
```diff
@@ -194,7 +198,7 @@ def _apply_shared_node_insertion_with_compression_type(
 
     @staticmethod
     def _apply_extraction_with_fused_bias_transformations(
```
Comment (daniil-lyakhov) [inline nncf/torch/model_transformer.py:200]:
```suggestion
    def _apply_extraction_transformations(
```

File: nncf/quantization/algorithms/fast_bias_correction/algorithm.py [hunk 4]
```diff
@@ -267,15 +270,18 @@ def output_filter_func(point):
...
 
-    def _extract_submodel(self, model_transformer: ModelTransformer, node_name: str) -> TModel:
+    def _extract_submodel(self, model_transformer: ModelTransformer, in_node_name: str, out_node_name: str) -> TModel:
         """
         Extracts sub-model using backend-specific ModelTransformer.
 
         :param model_transformer: Backend-specific ModelTransformer.
-        :param node_name: Name of the node that should be a center of the sub-model.
+        :param in_node_name: Name of the node that should be a center of the sub-model.
+        :param out_node_name: Name of the node that should be a center of the sub-model.
```
Comment (daniil-lyakhov) [inline nncf/quantization/algorithms/fast_bias_correction/algorithm.py:279]: Please update the description

File: nncf/quantization/algorithms/smooth_quant/torch_backend.py [hunk 5]
```diff
@@ -30,6 +30,8 @@
 from nncf.torch.graph.transformations.command_creation import create_command_to_update_weight
 from nncf.torch.graph.transformations.commands import PTSharedFnInsertionCommand
 from nncf.torch.graph.transformations.commands import PTTargetPoint
+from nncf.torch.model_graph_manager import get_const_data
+from nncf.torch.model_graph_manager import get_const_node
```
Comment (daniil-lyakhov) [inline nncf/quantization/algorithms/smooth_quant/torch_backend.py:34]: Duplicated name

**Comment by AlexanderDokuchaev** [comment]
manual/job/post_training_quantization/351/

**Review by daniil-lyakhov — APPROVED** [review]

File: nncf/quantization/algorithms/min_max/torch_backend.py [hunk 6]
```diff
@@ -206,7 +206,7 @@ def get_statistic_collector(
 
     @staticmethod
     def get_weight_tensor_port_ids(node: NNCFNode) -> List[Optional[int]]:
-        return [None]
+        return node.metatype.weight_port_ids
```
Comment (alexsu52) [inline nncf/quantization/algorithms/min_max/torch_backend.py:209]: As far as I understand, not all metatypes have a `weight_port_ids` attribute, then `[None]` should be returned, right?

**Review by alexsu52 — COMMENTED** [review]
LGTM. Could you provide performance numbers for QAT tests?

File: nncf/quantization/algorithms/min_max/torch_backend.py (hunk 6 above)
Comment (AlexanderDokuchaev) [reply nncf/quantization/algorithms/min_max/torch_backend.py:209]: Yes, but weighted nodes have attribute, it's filtered before using `get_weight_tensor_port_ids`
https://github.com/openvinotoolkit/nncf/blob/fa1a4ceabee06e10f7dfb42ce12bdd9b04ed4b79/nncf/quantization/algorithms/min_max/torch_backend.py#L353-L356

**Comment by AlexanderDokuchaev** [comment]
@alexsu52 
QAT - resnet18 | develop | PR | PR/dev
-- | -- | -- | --
total | 0:05:32 | 0:05:46 | 1.04
fine tune | 0:04:23 | 0:04:35 | 1.05

Model | Compr. Time (develop) | Stat. collection time (develop) | Compr. Time (PR) | Stat. collection time (PR) | CT PR/dev | SC PR/dev
-- | -- | -- | -- | -- | -- | --
timm/crossvit_9_240 | 0:00:45 | 0:00:41 | 0:00:55 | 0:00:52 | 1.22 | 1.27
timm/darknet53 | 0:00:43 | 0:00:40 | 0:00:45 | 0:00:42 | 1.05 | 1.05
timm/deit3_small_patch16_224 | 0:00:32 | 0:00:29 | 0:00:37 | 0:00:34 | 1.16 | 1.17
timm/dla34 | 0:00:24 | 0:00:21 | 0:00:26 | 0:00:24 | 1.08 | 1.14
timm/dpn68 | 0:00:47 | 0:00:42 | 0:00:52 | 0:00:48 | 1.11 | 1.14
timm/efficientnet_b0 | 0:00:38 | 0:00:34 | 0:00:45 | 0:00:41 | 1.18 | 1.21
timm/efficientnet_lite0 | 0:00:27 | 0:00:24 | 0:00:31 | 0:00:29 | 1.15 | 1.21
timm/hrnet_w18 | 0:03:54 | 0:03:22 | 0:04:58 | 0:04:38 | 1.27 | 1.38
timm/inception_resnet_v2 | 0:02:58 | 0:02:37 | 0:03:35 | 0:03:20 | 1.21 | 1.27
timm/levit_128 | 0:01:16 | 0:01:10 | 0:01:28 | 0:01:23 | 1.16 | 1.19
timm/mobilenetv2_050 | 0:00:27 | 0:00:24 | 0:00:31 | 0:00:29 | 1.15 | 1.21
timm/mobilenetv3_small_050 | 0:00:23 | 0:00:21 | 0:00:27 | 0:00:24 | 1.17 | 1.14
timm/regnetx_002 | 0:00:27 | 0:00:23 | 0:00:29 | 0:00:26 | 1.07 | 1.13
timm/resnest14d | 0:00:22 | 0:00:19 | 0:00:23 | 0:00:20 | 1.05 | 1.05
timm/resnet18 | 0:00:16 | 0:00:13 | 0:00:16 | 0:00:14 | 1.00 | 1.08
timm/swin_base_patch4_window7_224_no_sq | 0:01:27 | 0:01:19 | 0:01:44 | 0:01:38 | 1.20 | 1.24
timm/tf_inception_v3 | 0:01:07 | 0:01:00 | 0:01:13 | 0:01:08 | 1.09 | 1.13
timm/vgg11 | 0:00:33 | 0:00:29 | 0:00:31 | 0:00:29 | 0.94 | 1.00
timm/visformer_small | 0:01:04 | 0:01:00 | 0:01:08 | 0:01:03 | 1.06 | 1.05
timm/wide_resnet50_2 | 0:00:47 | 0:00:42 | 0:00:49 | 0:00:45 | 1.04 | 1.07

**Review by alexsu52 — APPROVED** [review]

**Pull request merged**